tests/foreman/ui/test_contenthost.py

```diff
@@ -1313,3 +1315,38 @@ def test_pagination_multiple_hosts_multiple_pages(session, module_host_template)
         # Assert that total items reported is the number of hosts created for this test
         total_items_found = pagination_values['total_items']
         assert int(total_items_found) >= host_num
+
+
+@tier3
+def test_search_for_virt_who_hypervisors(session):
+    """
+    Search the virt_who hypervisors with hypervisor=True or hypervisor=False.
+
+    :id: 3c759e13-d5ef-4273-8e64-2cc8ed9099af
+
+    :expectedresults: Search with hypervisor=True and hypervisor=False gives the correct result.
+
+    :BZ: 1653386
+
+    :CaseLevel: System
+
+    :CaseImportance: Medium
+    """
+    org = entities.Organization().create()
+    with session:
+        session.organization.select(org.name)
+        assert not session.contenthost.search("hypervisor = true")
+        # create virt-who hypervisor through the fake json conf
+        data = hypervisor_json_create(hypervisors=1, guests=1)
+        hypervisor_name = data['hypervisors'][0]['hypervisorId']
+        url = f"https://{settings.server.hostname}/rhsm/hypervisors/{org.label}"
+        auth = (settings.server.admin_username, settings.server.admin_password)
+        result = requests.post(url, auth=auth, verify=False, json=data)
+        assert result.status_code == 200
```

Comment:
Not blocking, but it might be good idea to extract this to it's own function in `virtwho_utils`. So in test you would only:
```python
data = hypervisor_json_create(hypervisors=1, guests=1)
create_fake_hypervisor_content(data)
```
Or even
```python
create_fake_hypervisor_content(hypervisors=1, guests=1)
```

Depending on whether there might be cases where we want to submit custom json, or are we always going to depend on `hypervisor_json_create`.

Reply:
Thank you for the suggestion. Will optimize this with next commit